### PR TITLE
[FSPE-7698] Fixed Chi Vue Data Table single page reactiveness

### PR DESCRIPTION
**Pull request by janibolkvadze**
@dani-cl-madrid @SergioQCL 

https://ctl.atlassian.net/browse/FSPE-7698

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://aa3ac1a440cba9ce1b68f6c948d3cb4b0f8ad0c5-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://9f267d988b8ed6d070838d6e0334f1e91d64364e-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 1]
```diff
@@ -267,15 +267,30 @@ export default class DataTable extends Vue {
     this.emitSelectedRows();
   }
 
+  calculateNumberOfPages() {
+    const numberOfPages = Math.ceil(this.data.body.length / this.resultsPerPage);
+
+    if (numberOfPages === 0) return 1;
+    return numberOfPages;
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/data-table/DataTable.tsx:274]:
```suggestion
    return Math.ceil(this.data.body.length / this.resultsPerPage) || 1;
```

Comment (dani-cl-madrid) [reply src/chi-vue/src/components/data-table/DataTable.tsx:274]: Or even
```Typescript
return Math.max(Math.ceil(this.data.body.length / this.resultsPerPage), 1);
```

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default class DataTable extends Vue {
   selectedRows: DataTableRow[] = [];
   slicedData: DataTableRow[] = [];
   sortable = false;
-  sortedData?: DataTableRow[];
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/data-table/DataTable.tsx:40]: If you decorate this property with `@State`, view will re-render when this property is updated.

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 3]
```diff
@@ -267,15 +267,30 @@ export default class DataTable extends Vue {
     this.emitSelectedRows();
   }
 
+  calculateNumberOfPages() {
+    const numberOfPages = Math.ceil(this.data.body.length / this.resultsPerPage);
+
+    if (numberOfPages === 0) return 1;
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:273]: I'd use {} just to improve readability

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 4]
```diff
@@ -267,15 +267,30 @@ export default class DataTable extends Vue {
...
+    return numberOfPages;
+  }
+
   selectAllRows(action: 'select' | 'deselect') {
+    const numberOfPages = this.calculateNumberOfPages();
+    const data =
+      numberOfPages === 1
+        ? this.sortedData && this.sortedData.length > 0
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:281]:
```suggestion
        ? this.sortedData?.length > 0
```

**Review by dani-cl-madrid — CHANGES_REQUESTED** [review]

File: src/chi-vue/src/components/data-table/DataTable.tsx (hunk 2 above)
Comment (janibolkvadze) [reply src/chi-vue/src/components/data-table/DataTable.tsx:40]: sortedData?: DataTableRow[] = []; <-- this is already reactive

Having an initial value and not having an underscore prefix (_property) change of property value is reactive

**Review by SergioQCL — APPROVED** [review]
ACK

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://3bc5226e69a9a262d0a451c6474c92f5a51ee4ec-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://e110e53403b24f16ad432f62907a5404f1494a05-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/5/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/6/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/7/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/9/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/11/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-37/12/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://37e2c63d292024b7c2f9cb4efc266641eeff495b-chi-pr.enterprise.ctl.io

**Review by SergioQCL — APPROVED** [review]
Go ahead!

**Pull request merged**